enterprise/server/raft/rangelease/rangelease.go

```diff
@@ -261,22 +261,13 @@ func (l *Lease) renewLease(ctx context.Context) error {
 	return nil
 }
 
-func (l *Lease) ensureValidLease(ctx context.Context, forceRenewal bool) (*rfpb.RangeLeaseRecord, error) {
+func (l *Lease) renewLeaseUntilValid(ctx context.Context) (*rfpb.RangeLeaseRecord, error) {
 	l.mu.Lock()
 	defer l.mu.Unlock()
 
 	ctx, cancel := context.WithTimeout(ctx, 3*time.Second)
 	defer cancel()
 
-	alreadyValid := false
-	if err := l.verifyLease(ctx, l.leaseRecord); err == nil {
-		alreadyValid = true
-	}
```

Comment:
do you need to gate the code in keepLeaseAlive behind the lease being not currently valid? Otherwise it seems like we'll be calling renewLeaseUntilValid() every time. Maybe that's ok, just checking?

Reply:
I think this is what we are currently doing by setting force-renewal to true.